api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/art/NCDDateDataEvaluator.java

```diff
@@ -34,11 +34,11 @@ public class NCDDateDataEvaluator implements PersonDataEvaluator {
     public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
         EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
 
-        String qry = "select ci.patient_id,\n" +
-                "       group_concat(ci.chronic_illness_onset_date) as ChronicIllness_onset\n" +
-                "from kenyaemr_etl.etl_allergy_chronic_illness ci\n" +
-                "where ci.visit_date <= date(:endDate)\n" +
-                "group by ci.patient_id;";
+        String qry = "SELECT ci.patient_id,\n" +
```

Comment:
You need to handle a case where there are more than on NCD recorded in that visit

Reply:
Resolved. Thanks